app/controllers/admin/users_controller.rb

```diff
@@ -97,10 +97,15 @@ def change_salesforce_contact
 
       begin
         contact = OpenStax::Salesforce::Remote::Contact.find(new_id) if check_really_exists
-        # if didn't explode, we found it, let cron job update other info
-        @user.salesforce_contact_id = new_id
-        return @user.save
+
+        if contact.present? || new_id.nil?
+          @user.salesforce_contact_id = new_id
+          return @user.save
+        end
       rescue
+        # exploded, probably due to badly formed SF ID
+      ensure
+        # either exploded or contact was `nil`
         flash[:alert] = "Can't find a Salesforce contact with ID #{new_id}"
```

Comment:
I think I'm trying to do too many things in one block.  I do want to save a nil `new_id` value (that's how I'm letting users clear a contact ID).  The `find` is also a bit funky in that it is a SF finder, not an ActiveRecord one, so it gives a weird exception if the ID isn't of the proper format, it returns `nil` if the ID is of the proper format but actually not found.  I bet if I split it up I can make it more clear.